librz/asm/asm.c

```diff
@@ -14,6 +18,69 @@
 
 RZ_LIB_VERSION(rz_asm);
 
+/**
+ * \brief Checks if the first character of \p c is a digit character
+ * OR if the first two chars are a hex prefix.
+ *
+ * \param c The character string.
+ * \return true First char is a digit or the first two chars are a hex prefix.
+ * \return false Otherwise.
+ */
+static bool is_num(const char *c) {
+	rz_return_val_if_fail(c, false);
+	if (!isascii(*c)) {
+		return false; // UTF-8
+	}
+	return rz_num_is_hex_prefix(c) || isxdigit(c[0]);
+}
+
+/**
+ * \brief Checks if the first character of \p c is an alphanumeric character OR if it is a hex prefix.
+ *
+ * \param c The character string.
+ * \return true If it is alphanumeric or a hex prefix.
+ * \return false Otherwise.
+ */
+static bool is_alpha_num(const char *c) {
+	rz_return_val_if_fail(c, false);
+	if (!isascii(*c)) {
+		return false; // UTF-8
+	}
+	return is_num(c) || isalpha(c[0]);
+}
+
+static bool is_separator(const char c) {
+	if (!isascii(c)) {
+		return false; // UTF-8
+	}
+	return (c == '(' || c == ')' || c == '[' || c == ']' || c == '{' || c == '}' || c == ',' || c == '.' || c == '#' || c == ':' || c == ' ');
```

Comment:
In TMS320 series of DSP, `||` is oftenly used to separate two parallel instructions in the block.

``` nasm
mov *(ar1+t0b) << t3, ac2 || mov ac1.l , *ar4
```

currently determined "mov ac1.l, *ar4" instruction as part of "ac2" operand

``` nasm
rptblocal #0x000011 || sfts ac0,  #-1
```

currently determined "sfts" as part of 0x000011 number

``` nasm
cmp t2 == t3, tc1
```

currently determined "t2 ==" as part of instruction keyword "cmp"

``` nasm
xccpart ac0 <=  #0 || mov #0x0, t0
```

currently determined "ac0" as part of "xccpart" instruction keyword, "|| mov #0x0, t0" as part of number #0

Reply:
I added `||`, `==` and `<=` to the separator list. But operators and separators are currently printed in the same color anyway.